perceval/runtime/job_group.py

```diff
@@ -0,0 +1,281 @@
+        else:
+            self._create_job_group()
+
+
+    @property
+    def name(self):
+        """
+        Name of the job Group [also the filename (without extension) on disk]
+        """
+        return self._name
+
+    @property
+    def job_group_data(self):
+        """
+        List of necessary information (id,status,metadata) for each job within the group
+        """
+        return self._group_info['job_group_data']
+
+    @staticmethod
+    def _get_current_datetime():
+        return datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    @staticmethod
+    def _get_job_group_dir():
+        return os.path.join(PersistentData().directory, "job_group")
+
+    @staticmethod
+    def _setup_job_group_dir(dir_path):
+        # creates a sub folder in persistent data directory if non-existent
+        try:
+            if not os.path.exists(dir_path):
+                os.makedirs(dir_path)
+        except OSError as exc:
+            warnings.warn(UserWarning(f"{exc}"))
+
+        # validate read/write permissions for the directory
+        if not os.access(dir_path, os.W_OK) or not os.access(dir_path, os.R_OK):
+            warnings.warn(UserWarning(f"Cannot read or write in {dir_path}"))
+
+    def _create_job_group(self):
+        # Saves information for a new job group on disk using PersistentData(),
+        jgrp_dir_path = JobGroup._get_job_group_dir()
+        JobGroup._setup_job_group_dir(jgrp_dir_path)  # create directory and validat permissions
+
+        self._group_info['created_date'] = JobGroup._get_current_datetime()
+        self._group_info['modified_date'] = JobGroup._get_current_datetime()
+        self._group_info['job_group_data'] = []
+        # write to disk
+        self._write_job_group_to_disk()
+
+    def _read_job_group_from_disk(self) -> dict:
+        # returns the Job group data stored on disk
+        jgrp_dir = JobGroup._get_job_group_dir()
+        ps = PersistentData()
+        group_data = json.loads(ps.read_file(os.path.join(jgrp_dir, self._file_name),
+                                             FileFormat.TEXT))
+        return group_data
+
+    def _write_job_group_to_disk(self):
+        # writes job group data to disk
+        jgrp_dir = JobGroup._get_job_group_dir()
+        PersistentData().write_file(os.path.join(jgrp_dir, self._file_name),
+                                    json.dumps(self._group_info), FileFormat.TEXT)
+
+    def _load_job_group(self):
+        # Creates a Job Group by loading an existing one from file
+        group_data = self._read_job_group_from_disk()
+
+        self._group_info['created_date'] = group_data['created_date']
+        self._group_info['modified_date'] = JobGroup._get_current_datetime()
+        self._group_info['job_group_data'] = group_data['job_group_data']
+        # Write to disk (necessary to update modification time)
+        self._write_job_group_to_disk()
+
+    @staticmethod
+    def list_saved_job_groups():
```

Comment:
missing typehinting for return object

Reply:
Also, the method name is a bit long, and is called by typing the class name anyway. So I'd suggest:
`JobGoup.list_existing()`